File: portal-kernel/src/com/liferay/portal/kernel/model/Address.java
/*
 * SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
 * SPDX-License-Identifier: LGPL-2.1-or-later
 */

package com.liferay.portal.kernel.model;

import com.liferay.portal.kernel.annotation.ImplementationClassName;
import com.liferay.portal.kernel.util.Accessor;

import org.osgi.annotation.versioning.ProviderType;

/**
 * The extended model interface for the Address service. Represents a row in the &quot;Address&quot; database table, with each column mapped to a property of this class.
 *
 * @author Brian Wing Shun Chan
 * @see AddressModel
 * @generated
 */
@ImplementationClassName("com.liferay.portal.model.impl.AddressImpl")
@ProviderType
public interface Address extends AddressModel, PersistedModel {

	/*
	 * NOTE FOR DEVELOPERS:
	 *
	 * Never modify this interface directly. Add methods to <code>com.liferay.portal.model.impl.AddressImpl</code> and rerun ServiceBuilder to automatically copy the method declarations to this interface.
	 */
	public static final Accessor<Address, Long> ADDRESS_ID_ACCESSOR =
		new Accessor<Address, Long>() {

			@Override
			public Long get(Address address) {
				return address.getAddressId();
			}

			@Override
			public Class<Long> getAttributeClass() {
				return Long.class;
			}

			@Override
			public Class<Address> getTypeClass() {
				return Address.class;
			}

		};

	public Country getCountry();

	public Region getRegion();

	public ListType getType();

}

File: modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/address.js
/*
 * SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
 * SPDX-License-Identifier: LGPL-2.1-or-later
 */

AUI.add(
	'liferay-address',
	() => {
		if (!Liferay.Address) {
			Liferay.Address = {
				getCountries(callback) {
					Liferay.Service(
						'/country/get-countries',
						{
							active: true,
						},
						callback
					);
				},

				getRegions(callback, selectKey) {
					Liferay.Service(
						'/region/get-regions',
						{
							active: true,
							countryId: Number(selectKey),
						},
						callback
					);
				},
			};
		}
	},
	'',
	{
		requires: [],
	}
);
